alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt

```diff
@@ -130,6 +131,23 @@ class AlertIndices(
         fun findingMapping() =
             AlertIndices::class.java.getResource("finding_mapping.json").readText()
 
+        @JvmStatic
+        fun getOrDefaultAlertIndex(monitor: Monitor): String {
+            var alertIndex = ALERT_INDEX
+            if (monitor.dataSources != null && monitor.dataSources.alertsIndex != null) {
+                alertIndex = monitor.dataSources.alertsIndex!!
+            }
+            return alertIndex
+        }
+
+        @JvmStatic
+        fun getOrDefaultFindingsHistoryIndex(monitor: Monitor): String {
+            if (monitor.dataSources?.findingsIndex == null || monitor.dataSources.findingsIndex!!.isEmpty()) {
+                return FINDING_HISTORY_WRITE_INDEX
+            }
+            return monitor.dataSources.findingsIndex!!
+        }
+
```

Comment:
the `getOrDefault` methods will not be needed anymore as `monitor.dataSources` will either have `custom values` or have `default values`.